File: packages/py/analytics/src/bolsa_analytics/cognitive/paper_broker.py
"""PaperBrokerReceipt — sello venue PAPER (ADR-034 post-OI-6).

PaperBroker = capa paper antes de BrokerAdapter.
≠ PaperOrder ≠ ExecutionRecord ≠ ExecutionPlan ≠ broker live.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal

from bolsa_analytics.cognitive.paper_order import PaperOrder

PaperBrokerVenue = Literal["PAPER"]
PaperBrokerAdapter = Literal["paper_broker"]
PaperBrokerFillStatus = Literal["executed", "unknown"]

PAPER_BROKER_KEY = "paperBroker"
PAPER_BROKER_ADAPTER: PaperBrokerAdapter = "paper_broker"


@dataclass(frozen=True, slots=True)
class PaperBrokerReceipt:
    """Foto honesta del submit paper. Venue siempre PAPER."""

    venue: PaperBrokerVenue
    adapter: PaperBrokerAdapter
    paper_order: PaperOrder
    fill_status: PaperBrokerFillStatus

    def to_dict(self) -> dict[str, object]:
        return {
            "venue": self.venue,
            "adapter": self.adapter,
            "paperOrder": self.paper_order.to_dict(),
            "fillStatus": self.fill_status,
        }


def build_paper_broker_receipt(
    *,
    paper_order: PaperOrder,
    fill_status: PaperBrokerFillStatus,
) -> PaperBrokerReceipt:
    """Receipt tras submit: executed o unknown (OI-3/OI-4)."""
    return PaperBrokerReceipt(
        venue="PAPER",
        adapter=PAPER_BROKER_ADAPTER,
        paper_order=paper_order,
        fill_status=fill_status,
    )


def paper_broker_venue_copy() -> str:
    """Copy de mesa: PaperBroker ≠ broker live."""
    return "Venue paper (≠ broker live)"

File: packages/shared/src/cognitive/paper-broker.ts
/**
 * PaperBrokerReceipt — sello de venue paper (ADR-034 post-OI-6).
 * PaperBroker = capa paper antes de BrokerAdapter.
 * ≠ PaperOrder ≠ ExecutionRecord ≠ ExecutionPlan ≠ broker live.
 */

import type { PaperOrderV1 } from "./paper-order.js";

export type PaperBrokerVenueV1 = "PAPER";

export type PaperBrokerAdapterV1 = "paper_broker";

export type PaperBrokerFillStatusV1 = "executed" | "unknown";

export type PaperBrokerReceiptV1 = {
  venue: PaperBrokerVenueV1;
  adapter: PaperBrokerAdapterV1;
  paperOrder: PaperOrderV1;
  fillStatus: PaperBrokerFillStatusV1;
};

export const PAPER_BROKER_KEY = "paperBroker";

export const PAPER_BROKER_ADAPTER = "paper_broker" as const;

export type BuildPaperBrokerReceiptInputV1 = {
  paperOrder: PaperOrderV1;
  fillStatus: PaperBrokerFillStatusV1;
};

/** Receipt honesto tras un submit paper (CREATED o FILLED). */
export function buildPaperBrokerReceipt(
  input: BuildPaperBrokerReceiptInputV1,
): PaperBrokerReceiptV1 {
  return {
    venue: "PAPER",
    adapter: PAPER_BROKER_ADAPTER,
    paperOrder: input.paperOrder,
    fillStatus: input.fillStatus,
  };
}

/** Copy de mesa: PaperBroker ≠ broker live. */
export function paperBrokerVenueCopy(): string {
  return "Venue paper (≠ broker live)";
}
